spec/acceptance/scotland-schema/lodge_spec.rb

```diff
@@ -0,0 +1,187 @@
+      expect(lodge_assessment(
+        assessment_body: valid_rdsap_xml,
+        accepted_responses: [403],
+        scopes: %w[assessment:lodge],
+        auth_data: {
+          scheme_ids: [scheme_id],
+        },
+        schema_name: "RdSAP-Schema-S-19.0",
+        migrated: true,
+      ).status).to eq(403)
+    end
+
+    it "has all expected data points present" do
+
+      expected_rdsap_data = {
+        "assessment_id"=>"0000-0000-0000-0000-0000",
+         "date_of_assessment"=>'2023-06-27',
+         "date_registered"=>'2023-06-27',
+         "type_of_assessment"=>"RdSAP",
+         "current_energy_efficiency_rating"=>79,
+         "postcode"=>"FK1 1XE",
+         "date_of_expiry"=>'2033-06-26',
+         "address_line1"=>"1 Some Street",
+         "address_line2"=>"",
+         "address_line3"=>"",
+         "address_line4"=>"",
+         "town"=>"Newkirk",
+         "scheme_assessor_id"=>"SPEC000000",
+         "opt_out"=>false,
+         "address_id"=>"LPRN-0000000000",
+         "migrated"=>true,
+         "cancelled_at"=>nil,
+         "not_for_issue_at"=>nil,
+         "created_at"=>'2021-06-21',
+         "hashed_assessment_id"=>"4af9d2c31cf53e72ef6f59d3f59a1bfc500ebc2b1027bc5ca47361435d988e1a",
+         "test_column"=>nil
+      }
+
+      response = lodge_assessment assessment_body: valid_rdsap_xml,
+                                  accepted_responses: [201],
+                                  scopes: %w[assessment:lodge migrate:assessment],
+                                  auth_data: {
+                                    scheme_ids: [scheme_id],
+                                  },
+                                  schema_name: "RdSAP-Schema-S-19.0",
+                                  migrated: "true"
+
+      expect(JSON.parse(response.body, symbolize_names: true)[:data][:assessments].first).to eq "0000-0000-0000-0000-0000"
+      expect(migrated_scotland_rdsap_data).to eq expected_rdsap_data
+    end
+
+    context "when migrating the same assessment ID" do
+      before do
+        lodge_assessment assessment_body: valid_rdsap_xml,
+                         accepted_responses: [201],
+                         scopes: %w[assessment:lodge migrate:assessment],
+                         auth_data: {
+                           scheme_ids: [scheme_id],
+                         },
+                         schema_name: "RdSAP-Schema-S-19.0",
+                         migrated: "true"
+      end
+
+      it "is true in migrated column" do
+        lodge_assessment assessment_body: valid_rdsap_xml,
+                         accepted_responses: [201],
+                         scopes: %w[assessment:lodge migrate:assessment],
+                         auth_data: {
+                           scheme_ids: [scheme_id],
+                         },
+                         schema_name: "RdSAP-Schema-S-19.0",
+                         migrated: "true"
+        expect(migrated_scotland_rdsap_data["migrated"]).to be_truthy
```

Comment:
Not sure what the purpose is of this test is? 
Is it to show that you can migrate a assessment that is already there?
If you remove the before block this test will still pass.
Would a better expectation show a change a different value in the schema?

Reply:
This is meant to show that when you migrate you can overwrite an assessment with an existing RRN. It should error if migration is false. I have changed the test type to make it clearer.